service/src/main/java/bio/terra/cbas/initialization/cloneRecovery/CloneRecoveryTransactionService.java

```diff
@@ -1,37 +1,45 @@
 package bio.terra.cbas.initialization.cloneRecovery;
 
+import bio.terra.cbas.dao.MethodDao;
+import bio.terra.cbas.dao.MethodVersionDao;
 import bio.terra.cbas.dao.RunDao;
 import bio.terra.cbas.dao.RunSetDao;
 import bio.terra.cbas.models.RunSet;
 import bio.terra.common.db.WriteTransaction;
-import java.util.UUID;
 import org.springframework.stereotype.Service;
 
 @Service
 public class CloneRecoveryTransactionService {
   private final RunDao runDao;
   private final RunSetDao runSetDao;
+  private final MethodDao methodDao;
+  private final MethodVersionDao methodVersionDao;
 
-  public CloneRecoveryTransactionService(RunSetDao runSetDao, RunDao runDao) {
+  public CloneRecoveryTransactionService(
+      RunSetDao runSetDao, RunDao runDao, MethodDao methodDao, MethodVersionDao methodVersionDao) {
     this.runSetDao = runSetDao;
     this.runDao = runDao;
+    this.methodDao = methodDao;
+    this.methodVersionDao = methodVersionDao;
   }
 
-  private void deleteRuns(UUID runSetId) {
+  private void deleteRuns(RunSet runSet) {
     runDao
-        .getRuns(new RunDao.RunsFilters(runSetId, null))
+        .getRuns(new RunDao.RunsFilters(runSet.runSetId(), null))
         .forEach(r -> runDao.deleteRun(r.runId()));
+    methodDao.unsetLastRunSetId(runSet.methodVersion().getMethodId());
+    methodVersionDao.unsetLastRunSetId(runSet.getMethodVersionId());
```

Comment:
Nit: This is kind of a surprise addition to `deleteRuns` IMO. Today we only use it in two known places but if we re-use this in the future it might surprise us that it's also unsetting last run sets. 

Perhaps this belongs in a separate `unsetLastRunSetIdRecords` function?

Reply:
ah, I hadn't considered that. I'll move it somewhere else